src/SqlBindingUtilities.cs

```diff
@@ -290,5 +290,51 @@ internal static bool IsFatalSqlException(this Exception e)
             }
             return true;
         }
+
+        /// <summary>
+        /// Query the ServerProperty('EngineEdition') for logging the Server Type for Telemetry
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Throw if an error occurs while querying the engine edition</exception>
+        public static async Task<string> GetSqlServerEdition(SqlConnection connection, ILogger logger, CancellationToken cancellationToken)
+        {
+            string selectServerEdition = $"SELECT ServerProperty('EngineEdition')";
```

Comment:
We should also get the `SERVERPROPERTY('Edition')` to get the server edition as well (Azure vs On-Prem mostly)

Reply:
Edition returns the below list, not sure if that'll be any more insightful than EngineEdition:
![image](https://user-images.githubusercontent.com/12754347/215866331-7da4ec70-71d0-4db4-a889-3fd52a402bc5.png)

Reply:
Edition != Engine Edition. They are separate things, engine edition is per database and edition is per server. Both are useful to have.

Reply:
So Edition = Server edition and EngineEdition = Database Engine edition.
@chlafreniere  @dzsquared Want me to add both now?

Reply:
Added it, we can clean up if it's not required.